### PR TITLE
Map netflows to accessible context key values

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
Ready

## Related Issues
fixes: demisto/etc#13615

## Description
Copied netflow outputs to anticipated context key values.

Left old keys for backwards compatibility

```
{
    "Protectwise": {
        "Event": {
            "confidence": 100,
            "endedAt": "2019-03-06T11:24:45.466Z",
            "killChainStage": "Cnc",
            "netflowCount": 2,
            "Netflows": [{
                "dstIp": "192.168.10.90",
                "key": "0000016952899fd5951b2cc79bc30e23",
                "id.srcPort": 55381,
                "srcPort": 55381,
                "id.dstPort": 80,
                "id.srcIp": "115.78.165.19",
                "id.dstIp": "192.168.10.90",
                "id.layer4Proto": "Tcp",
                "srcIp": "115.78.165.19",
                "dstPort": 80,
                "layer4Proto": "Tcp"
            }, {
                "dstIp": "192.168.10.90",
                "key": "0000016952bfa2cd951b2cc79bc30e23",
                "id.srcPort": 55381,
                "srcPort": 55381,
                "id.dstPort": 80,
                "id.srcIp": "115.78.165.19",
                "id.dstIp": "192.168.10.90",
                "id.layer4Proto": "Tcp",
                "srcIp": "115.78.165.19",
                "dstPort": 80,
                "layer4Proto": "Tcp"
            }],
            "threatScore": 65,
            "observedStage": "Realtime",
            "Observations": [{
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T10:26:06.503Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T10:25:45.762Z",
                "netflowId": "0000016952899fd5951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952899fd5951b2cc79bc30e23651e37b20000",
                "killChainStage": "Cnc"
            }, {
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T11:25:05.045Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T11:24:45.104Z",
                "netflowId": "0000016952bfa2cd951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952bfa2cd951b2cc79bc30e2364e654980000",
                "killChainStage": "Cnc"
            }, {
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T11:25:05.045Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T11:24:45.466Z",
                "netflowId": "0000016952bfa2cd951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952bfa2cd951b2cc79bc30e234b72aa350000",
                "killChainStage": "Cnc"
            }, {
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T10:26:06.503Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T10:25:45.762Z",
                "netflowId": "0000016952899fd5951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952899fd5951b2cc79bc30e23c475b9610000",
                "killChainStage": "Cnc"
            }, {
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T10:26:06.503Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T10:25:45.400Z",
                "netflowId": "0000016952899fd5951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952899fd5951b2cc79bc30e23f53a68cc0000",
                "killChainStage": "Cnc"
            }, {
                "confidence": 100,
                "endedAt": null,
                "observedAt": "2019-03-06T11:25:05.045Z",
                "severity": 65,
                "threatScore": 65,
                "observedStage": "Realtime",
                "source": "Surricata",
                "occurredAt": "2019-03-06T11:24:45.466Z",
                "netflowId": "0000016952bfa2cd951b2cc79bc30e23",
                "sensorId": 3110,
                "id": "0000016952bfa2cd951b2cc79bc30e2315b7b5130000",
                "killChainStage": "Cnc"
            }],
            "startedAt": "2019-03-06T10:25:45.400Z",
            "sensorId": 3110,
            "message": "Probable C2 Activity on Host: 192.168.10.90",
            "observationCount": 6,
            "type": "MaliciousConversation",
            "id": "0005836a699fb6c073069e8cbc4400f72e1fac7a528e8cc00879c147",
            "observedAt": "2019-03-06T11:25:15.247Z"
        }
    }
}
```

## Required version of Demisto
3.5.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/integration-ProtectWise.yml [hunk 1]
```diff
@@ -299,7 +299,14 @@ script:
...
             if (res.netflows && res.netflows.length > 0) {
                 var nf = filterClone(res.netflows, DEFAULT_NETFLOW_HEADERS);
-                event['Netflows'] = nf;
+                for(var i = 0; i < nf.length; i++) {
+                    nf[i]['srcIp'] = nf[i]['id.srcIp'];
+                    nf[i]['dstIp'] = nf[i]['id.dstIp'];
+                    nf[i]['srcPort']  = nf[i]['id.srcPort'];
+                    nf[i]['dstPort'] = nf[i]['id.dstPort'];
+                    nf[i]['layer4Proto'] = nf[i]['id.layer4Proto'];
+                    event['Netflows'] = nf;
```
Comment (yardensade) [inline Integrations/integration-ProtectWise.yml]: Why not update event *after* the loop ends? 🤔

File: Integrations/integration-ProtectWise.yml [hunk 2]
```diff
@@ -390,6 +397,14 @@ script:
             var md = '### Protectwise PCAP for Event ' + res.id + '\n - Estimated size: ' + res.estimatedSize + ' bytes';
             if (res.netflows && res.netflows.length > 0) {
                 var nf = filterClone(res.netflows, DEFAULT_NETFLOW_HEADERS);
+                    for(var i = 0; i < nf.length; i++) {
+                        nf[i]['srcIp'] = nf[i]['id.srcIp'];
+                        nf[i]['dstIp'] = nf[i]['id.dstIp'];
+                        nf[i]['srcPort']  = nf[i]['id.srcPort'];
+                        nf[i]['dstPort'] = nf[i]['id.dstPort'];
+                        nf[i]['layer4Proto'] = nf[i]['id.layer4Proto'];
+                        event['Netflows'] = nf;
```
Comment (yardensade) [inline Integrations/integration-ProtectWise.yml]: same here <

File: Integrations/integration-ProtectWise.yml [hunk 3]
```diff
@@ -413,6 +428,14 @@ script:
             var md = '### Protectwise PCAP for Observation ' + res.id + '\n - Estimated size: ' + res.estimatedSize + ' bytes';
             if (res.netflows && res.netflows.length > 0) {
                 var nf = filterClone(res.netflows, DEFAULT_NETFLOW_HEADERS);
+                    for(var i = 0; i < nf.length; i++) {
+                        nf[i]['srcIp'] = nf[i]['id.srcIp'];
+                        nf[i]['dstIp'] = nf[i]['id.dstIp'];
+                        nf[i]['srcPort']  = nf[i]['id.srcPort'];
+                        nf[i]['dstPort'] = nf[i]['id.dstPort'];
+                        nf[i]['layer4Proto'] = nf[i]['id.layer4Proto'];
+                        event['Netflows'] = nf;
```
Comment (yardensade) [inline Integrations/integration-ProtectWise.yml]: And here as well

File: Integrations/integration-ProtectWise.yml (hunk 1 above)
Comment (amshamah419) [reply Integrations/integration-ProtectWise.yml]: Good point... Fixed ;)

**Comment by yardensade** [comment]
If the 2 scripts are not testable it is better to write a reason, so someone in the future won't have to wonder about that too much if/when he updates it.

**Comment by amshamah419** [comment]
Just added tests for the scripts :)

**Pull request merged**